api/v1/dspipeline_types.go

```diff
@@ -151,6 +151,12 @@ type APIServer struct {
 	// +kubebuilder:default:=60
 	// +kubebuilder:validation:Optional
 	ArtifactSignedURLExpirySeconds *int `json:"artifactSignedURLExpirySeconds"`
+
+	// The storage for pipeline definitions (pipelines and pipeline versions). It can be
+	// either 'database' or 'kubernetes' (Pipeline and PipelineVersion kinds). Defaults to 'database'.
+	// +kubebuilder:default:=database
+	// +kubebuilder:validation:Optional
```

Comment:
This will enforce that only valid values are set on this field at the Kubernetes API level (e.g. reject invalid values when doing kubectl apply).
```suggestion
	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Enum=database;kubernetes
```